tests/linalg_test.py

```diff
@@ -42,6 +42,29 @@
 float_types = [np.float32, np.float64]
 complex_types = [np.complex64, np.complex128]
 
+
+# Used in tests for det and lu
+singular_mats = [
+  jnp.array([[ 50, -30,  45],  # rank-2, corank-1
+             [-30,  90, -81],
+             [ 45, -81,  81]], dtype=jnp.float32),
+  jnp.array([[ 36, -42,  18],  # rank-1, corank-2
+             [-42,  49, -21],
+             [ 18, -21,   9]], dtype=jnp.float32),
+]
+
+
+nonsquare_singular_mats = [
+  jnp.array([[-35,   7,  27, -17],
+             [ 11, -13,  -9,  11],
+             [ 19, -11, -15,  13]], dtype=jnp.float32),
+  jnp.array([[-35,  11,  19],
+             [  7, -13, -11],
+             [ 27,  -9, -15],
+             [-17,  11,  13]], dtype=np.float32)
+]
```

Comment:
This PR broke internal tests because the device is not yet initialized when these lines are run. We need to replace these module-level constants with functions.

Reply:
You can also often use classic NumPy arrays in this situation.

Reply:
Should be fixed by #3684